src/env.rs

```diff
@@ -1,13 +1,20 @@
 use libc::c_char;
 use pam_sys::{getenvlist, raw, PamHandle};
 
+use core::iter::FusedIterator;
 use std::ffi::CStr;
 
 pub struct PamEnvList {
     ptr: *const *const c_char,
 }
 
-pub fn get_pam_env(handle: &mut PamHandle) -> Option<PamEnvList> {
+pub struct IterPamEnv<'a> {
```

Comment:
Again to follow conventions, I think this should be called `PamEnvIter`. 

```suggestion
pub struct PamEnvIter<'a> {
```